tpm/src/socket.rs

```diff
@@ -0,0 +1,166 @@
+        self.connect(&path)?;
+        Ok(())
+    }
+
+    pub fn connect(&mut self, socket_path: &str) -> Result<()> {
+        self.state = SocDevState::Connecting;
+
+        let now = Instant::now();
+
+        // Retry connecting for 10 secs
+        loop {
+            match UnixStream::connect(socket_path) {
+                Ok(s) => {
+                    let fd = s.as_raw_fd();
+                    self.ctrl_fd = fd;
+                    self.stream = Some(s);
+                    self.state = SocDevState::Connected;
+                    debug!("Connected to TPM socket path : {:?}", socket_path);
+                    return Ok(());
+                }
+                Err(_e) => {}
+            };
+            sleep(Duration::from_millis(1000));
+
+            if now.elapsed().as_secs() >= 10 {
+                break;
+            }
+        }
+        Err(TPMSocError::ConnectToTPMSocket(anyhow!(
+            "Failed to connect to TPM Socket"
+        )))
+    }
+
+    pub fn set_datafd(&mut self, fd: RawFd) {
+        self.data_fd = fd;
+    }
+
+    pub fn set_msgfd(&mut self, fd: RawFd) {
+        self.write_msgfd = fd;
+    }
+
+    pub fn send_full(&self, buf: &[u8]) -> Result<usize> {
+        let write_fd = self.write_msgfd;
+
+        let size = self
+            .stream
+            .as_ref()
+            .unwrap()
+            .send_with_fd(buf, write_fd)
+            .map_err(|e| {
+                TPMSocError::WriteTPMSocket(anyhow!(
+                    "Failed to write to TPM Socket. Error: {:?}",
+                    e
+                ))
+            })?;
+
+        Ok(size as usize)
+    }
+
+    pub fn write(&mut self, buf: &[u8]) -> Result<usize> {
+        if self.stream.is_none() {
+            return Err(TPMSocError::WriteTPMSocket(anyhow!(
+                "Stream for TPM Socket was not initialized"
+            )));
+        }
+        let res = match self.state {
+            SocDevState::Connected => {
+                let ret = self.send_full(buf)?;
+                // swtpm will receive data Fd after a successful send
+                // Reset cached write_msgfd after a successful send
+                // Ideally, write_msgfd is reset after first Ctrl Command
+                if ret > 0 && self.write_msgfd != 0 {
+                    self.write_msgfd = 0;
+                }
+                ret
+            }
+            _ => {
+                return Err(TPMSocError::WriteTPMSocket(anyhow!(
+                    "TPM Socket was not in Connected State"
+                )))
+            }
+        };
+        Ok(res)
+    }
+
+    pub fn read(&mut self, buf: &mut [u8]) -> Result<usize> {
+        let newbuf: &mut [u8] = &mut [0; TPM_CRB_BUFFER_MAX];
+
+        if self.stream.is_none() {
+            return Err(TPMSocError::ReadTPMSoc(anyhow!(
+                "Stream for TPM Socket was not initialized"
+            )));
+        }
+        let mut sock = self.stream.as_ref().unwrap();
```

Comment:
Because here you shortened it to "sock".

Reply:
fixed all the shortened names.